cli: panic when adding registry which includes pack with malformed metadata

Internal report that when adding a registry which includes a pack with malformed or missing metadata, the command panics. I am not sure if this is related directly to add, or potentially to the cache ensure process.

```
$ nomad-pack registry add burkey https://github.com/pandom/nomad-packs
go-getter URL is https://github.com/pandom/nomad-packs
Registry successfully cloned at /Users/burkey/.nomad/packs/nomad-pack-tmp
Processing pack entries at /Users/burkey/.nomad/packs/nomad-pack-tmp
found pack entry minecraft
Processing pack minecraft@latest
Updating pack
Removing previous latest
Writing pack to /Users/burkey/.nomad/packs/burkey/minecraft@latest
Loading cloned pack from /Users/burkey/.nomad/packs/burkey/minecraft@latest
Calculating SHA for latest
  temp directory deleted
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x187f02f]
```

--- a/internal/pkg/cache/registry.go
+++ b/internal/pkg/cache/registry.go
@@ -133,8 +133,13 @@ func (r *Registry) parsePackURL(packURL string) bool {
 // setURLFromPacks sets the Source since we don't have this stored in any sort of
 // reliable way.
 func (r *Registry) setURLFromPacks() {
+
 	for _, cachedPack := range r.Packs {
-		if !r.parsePackURL(cachedPack.Metadata.Pack.URL) {
+		if err := cachedPack.Validate(); err != nil {
+			continue
+		}
+
+		if r.parsePackURL(cachedPack.Metadata.Pack.URL) {
 			continue
 		}
 

--- a/sdk/pack/metadata.go
+++ b/sdk/pack/metadata.go
@@ -115,11 +115,17 @@ func (md *Metadata) Validate() error {
 // validate the MetadataApp object to ensure it meets requirements and doesn't
 // contain invalid or incorrect data.
 func (ma *MetadataApp) validate() error {
+	if ma == nil {
+		return errors.New("App is uninitialized")
+	}
 	return nil
 }
 
 // validate the MetadataPack object to ensure it meets requirements and doesn't
 // contain invalid or incorrect data.
 func (mp *MetadataPack) validate() error {
+	if mp == nil {
+		return errors.New("Pack metadata is uninitialized")
+	}
 	return nil
 }

